src/react-apps/applications/shared/src/version-control/versionControlHeader.tsx

```diff
@@ -0,0 +1,392 @@
+  changesInLocalRepo: boolean;
+  moreThanAnHourSinceLastPush: boolean;
+  hasPushRight: boolean;
+  anchorEl: any;
+  modalState: any;
+  mergeConflict: boolean;
+}
+
+const theme = createMuiTheme(altinnTheme);
+
+const styles = createStyles({
+  headerStyling: {
+    background: theme.altinnPalette.primary.greyLight,
+    paddingTop: 10,
+  },
+});
+
+const initialState = {
+  header: '',
+  descriptionText: '',
+  isLoading: '',
+  shouldShowDoneIcon: false,
+  btnText: '',
+  shouldShowCommitBox: false,
+  btnMethod: '',
+};
+
+class VersionControlHeader extends React.Component<IVersionControlHeaderProps, IVersionControlHeaderState> {
+  public interval: any;
+  public _isMounted = false;
+  constructor(_props: IVersionControlHeaderProps) {
+    super(_props);
+    this.state = {
+      changesInMaster: false,
+      changesInLocalRepo: false,
+      moreThanAnHourSinceLastPush: false,
+      hasPushRight: false,
+      anchorEl: null,
+      mergeConflict: false,
+      modalState: initialState,
+    };
+  }
+
+  public performAPIFetch(url: string, callbackFunc: any, object?: any) {
+    fetch(url, object)
+      .then((response) => {
+        return response.text().then((text: any) => {
+          return text ? JSON.parse(text) : {};
+        });
+      })
+      .then(
+        (result) => {
+          callbackFunc(result);
+        },
+        (error) => {
+          console.error('Something went wrong: ' + error);
+        },
+      );
+  }
+
+  public getStatus(callbackFunc?: any) {
+    const altinnWindow: any = window as any;
+    const { org, service } = altinnWindow;
+    const url = `${altinnWindow.location.origin}/designerapi/Repository/RepoStatus?owner=${org}&repository=${service}`;
+    this.performAPIFetch(url, (result: any) => {
+      if (this._isMounted) {
+        if (callbackFunc) {
+          callbackFunc(result);
+        } else {
+          if (result) {
+            this.setState({
+              changesInMaster: result.behindBy !== 0,
+              changesInLocalRepo: result.contentStatus.length > 0,
+            });
+          }
+        }
+      }
+    });
+  }
+
+  public updateStateOnIntervals() {
+    this.getStatus();
+    this.getLastPush();
+  }
+
+  public getLastPush() {
+    if (!this.state.moreThanAnHourSinceLastPush) {
+      const altinnWindow: any = window as any;
+      const { org, service } = altinnWindow;
+      // tslint:disable-next-line:max-line-length
+      const url = `${altinnWindow.location.origin}/designerapi/Repository/GetLatestCommitFromCurrentUser?owner=${org}&repository=${service}`;
+      this.performAPIFetch(url, (result: any) => {
+        if (this._isMounted && result) {
+          const diff = new Date().getTime() - new Date(result.commiter.when).getTime();
+          const oneHour = 60 * 60 * 1000;
+          this.setState({
+            moreThanAnHourSinceLastPush: oneHour < diff,
+          });
+        }
+      });
+    }
+  }
+
+  public componentWillMount() {
```

Comment:
componentWillMount is deprecated and will be removed in future versions of React: https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

Have you tried to use componentDidMount instead?

Reply:
Using componentDidMount instead